src/class_sqlThread.py

```diff
@@ -487,154 +219,239 @@ def run(self):  # pylint: disable=too-many-locals, too-many-branches, too-many-s
+        self.db = None
+        self.max_setting_level = 4
+        self.rowcount = 0
+        logger.debug('Init thread in sqlthread')
+
+    @property
+    def sql_config_settings_version(self):
+        """ Getter for BMConfigParser (obj) """
+
+        return config.getint(
+            'bitmessagesettings', 'settingsversion')
+
+    @sql_config_settings_version.setter
+    def sql_config_settings_version(self, settingsversion):  # pylint: disable=R0201, no-self-use
+        # Setter for BmConfigparser
+
+        config.set(
+            'bitmessagesettings', 'settingsversion', str(int(settingsversion) + 1))
+        return config.save()
+
+    def upgrade_config_setting_version(self):
+        """
+            upgrade config parser setting version.
+            If the settings version is equal to 2 or 3 then the
+            sqlThread will modify the pubkeys table and change
+            the settings version to 4.
+        """
+        while self.sql_config_settings_version < self.max_setting_level:
+            self.db.upgrade_config_parser_setting_version(self.sql_config_settings_version)
+            self.sql_config_settings_version = self.sql_config_settings_version
+
+    @staticmethod
+    def error_handler(err, command):
+        if str(err) == 'database or disk is full':
+            logger.fatal(
+                "(While %s) Alert: Your disk or data storage volume is full. sqlThread will now exit.", command
+            )
+            queues.UISignalQueue.put((
+                'alert', (
+                    _translate(
+                        "MainWindow",
+                        "Disk full"),
+                    _translate(
+                        "MainWindow",
+                        'Alert: Your disk or data storage volume is full. Bitmessage will now exit.'),
+                    True)))
+            os._exit(0)
+
+    def is_query_commit(self):
+        """
+            When query == 'commit'
+        """
+        try:
+            self.db.conn.commit()
+        except Exception as err:
+            command = 'committing'
+            self.error_handler(err, command)
+
+    def is_query_movemessagstoprog(self):
+        """
+            When query == 'movemessagstoprogs'
+        """
+        try:
+            self.db.conn.commit()
+        except Exception as err:
+            command = 'movemessagstoprog'
+            self.error_handler(err, command)
+        self.db.conn.close()
+        shutil.move(
+            paths.lookupAppdataFolder() + 'messages.dat', paths.lookupExeFolder() + 'messages.dat')
+        self.db.conn = sqlite3.connect(paths.lookupExeFolder() + 'messages.dat')
+        self.db.conn.text_factory = str
+        self.db.cur = self.db.conn.cursor()
+
+    def is_query_deleteandvacuume(self):
+        """
+            When query == 'deleteandvacuume'
+        """
+        try:
+            self.db.cur.execute(''' VACUUM ''')
+        except Exception as err:
+            command = 'deleteandvacuume'
```

Comment:
I don't know if there needs to be a variable `command, why not just use the string directly?